packages/docs/jobs/manual-resolution.md

```diff
@@ -0,0 +1,279 @@
+---
+outline: deep
+title: Manual Job Resolution
+description: Manual Job Resolution guide for Sidequest.js
+---
+
+# Manual Job Resolution
+
+Manual Job Resolution is a deployment and bundling strategy for Sidequest.js that gives you complete control over how job classes are resolved and imported at runtime. Instead of relying on [automatic script path detection](/jobs/class#job-script-detection), you can manually specify which job classes are available through a central registry file.
+
+## What is Manual Job Resolution?
+
+By default, Sidequest.js automatically detects and imports job scripts based on their file paths. While this works well for development, it can cause issues in production environments with:
+
+- **Bundled applications** (webpack, esbuild, rollup, etc.)
+- **Containerized deployments** where file paths may change
+- **Serverless functions** with restricted file system access
+- **Complex monorepo structures** with dynamic import paths
+
+Manual Job Resolution solves these issues by using a single `sidequest.jobs.js` file that explicitly exports all available job classes.
+Therefore, you can compile your jobs separately and re-export them wherever they are needed.
+
+Moreover, this approach clears the chain of `../../../../` relative imports in the job scripts, which can be hard to manage and debug in large projects.
+When you see `manual-resolution` as the job script, it indicates that the job class will be resolved from the central registry file instead of a specific file path.
+
+## How It Works
+
+When manual job resolution is enabled:
+
+1. **Job Enqueuing**: Jobs are enqueued with `script: "manual-resolution"` instead of specific file paths
+2. **Job Execution**: The runner looks for a `sidequest.jobs.js` file in the current directory or any parent directory
+3. **Class Resolution**: Job classes are imported from the central registry file instead of individual script files
+
+## Setting Up Manual Job Resolution
+
+### Step 1: Enable Manual Resolution
+
+Configure your Sidequest engine to use manual job resolution:
+
+```typescript
+import { Sidequest } from "@sidequest/engine";
+
+// or Sidequest.configure
+await Sidequest.start({
+  backend: { driver: "@sidequest/sqlite-backend" },
+  queues: [{ name: "default" }],
+  manualJobResolution: true, // Enable manual job resolution
```

Comment:
is it possible check the existence of the `sidequest.jobs.js` file and automatically set it to true? That way we don't need a flag just adding the file to the project root would enable the feature.

Reply:
I think we should allow the user to control whether to enable/disable the feature with a flag